### ISSUE TITLE
OrderedClassElementsFixer - ClassAttributeSeparationFixer - priority

### ISSUE THREAD

**Issue by dmvdbrugge** [FriendsOfPHP/PHP-CS-Fixer#4444]
For configuration or updating questions please read the README and UPGRADE documentation,
or visit: https://gitter.im/PHP-CS-Fixer

When reporting an issue (bug) please provide the following information:

#### The PHP version you are using (`$ php -v`):
```
PHP 7.1.23 (cli) (built: Oct 11 2018 16:21:11) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.1.23, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.6.0, Copyright (c) 2002-2018, by Derick Rethans
```

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
```
PHP CS Fixer 2.15.1 Europe Round by Fabien Potencier and Dariusz Ruminski
```

#### The command you use to run PHP CS Fixer:
```
vendor/bin/php-cs-fixer fix -vvv --dry-run --diff
```

#### If applicable, please provide minimum samples of PHP code (as plain text, not screenshots):

 * with unexpected changes applied when running PHP CS Fixer:
```diff
 class Registration
 {
+    /**
+     * @var bool
+     */
+    protected $should_send_notifications = true;
     /**
      * @var \Database\MainDatabase
      */
@@ -51,11 +55,6 @@ class Registration
      */
     private $user_service;

-    /**
-     * @var bool
-     */
-    protected $should_send_notifications = true;
-
     /**
      * @var EmailQueue
      */
```

 * with the changes you expected instead:
```diff
 class Registration
 {
+    /**
+     * @var bool
+     */
+    protected $should_send_notifications = true;
+
     /**
      * @var \Database\MainDatabase
      */
@@ -51,11 +55,6 @@ class Registration
      */
     private $user_service;

-    /**
-     * @var bool
-     */
-    protected $should_send_notifications = true;
-
     /**
      * @var EmailQueue
      */
```

`OrderedElementsFixer` should run before `ClassAttributeSeparationFixer`, so the newline would've been inserted.

As the priority of `ClassAttributeSeparationFixer` is very high though, it might be easier for `OrderedElementsFixer` to take surrounding whitespace into account, and move it with it.

After writing up this report, I notice that `no_blank_lines_after_class_opening` triggered as well, so that probably means `OrderedElementsFixer` did take the newline before it into account, just not the one after it.

**Comment by SpacePossum** [comment]
>OrderedElementsFixer should run before ClassAttributeSeparationFixer, so the newline would've been inserted.

I'm pretty sure this has been the case for a long time;

https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.15.0/tests/Fixtures/Integration/priority/ordered_class_elements,class_attributes_separation.test

It also runs before `no_blank_lines_after_class_opening`.

Can you share an example that show the issue?

**Comment by dmvdbrugge** [comment]
TLDR: config.

I think this was a small oversight when I moved from fully self-defined config to using `@Symfony` as base, as Symfony has only `method` as default config for `class_attributes_separation`. It thus makes total sense what happens.

My bad 😳 issue closed.

**Comment by SpacePossum** [comment]
thanks for reporting back and seeing this has been resolved  :+1:  :)
